pkg/provision/migrations_202203_rbac_resource_fix.go

```diff
@@ -118,149 +206,170 @@ func migratePost202203RbacRule(r *rbac.Rule, rx *resIndex) (op int) {
 
 	if invalid {
 		op = 1
+	} else {
+		op = 0
+	}
 
-		partsLen := len(parts)
-		ID := uint64(0)
-		p1 := uint64(0)
-		p2 := uint64(0)
-
-		if partsLen == 1 {
-			ID = parseUint(parts[0])
-		} else if partsLen == 2 {
-			p1 = parseUint(parts[0])
-			ID = parseUint(parts[1])
-		} else if partsLen == 3 {
-			p1 = parseUint(parts[0])
-			p2 = parseUint(parts[1])
-			ID = parseUint(parts[2])
-		}
+	return
+}
 
-		rType := rbac.ResourceType(r.Resource)
-		switch rType {
-		case composeTypes.ModuleFieldResourceType:
-			if ID > 0 {
-				if f, ok := rx.fields[ID]; ok {
-					p2 = f.ModuleID
-					if m, ok := rx.modules[p2]; ok {
-						p1 = m.NamespaceID
-						if !validID(p1) {
-							return -1
-						}
-					} else {
-						return -1
-					}
-				} else {
-					return -1
-				}
-			}
+//  0 - no action
+// -1 - remove
+//  1 - update
+func migratePost202203RbacRule(r *rbac.Rule, rx *resIndex) (op int) {
+	op = 1
+
+	// split the IDs
+	parts := strings.SplitN(r.Resource, "/", 4)
+	parts = parts[1:]
 
-			r.Resource = composeTypes.ModuleFieldRbacResource(p1, p2, ID)
+	partsLen := len(parts)
+	ID := uint64(0)
+	p1 := uint64(0)
+	p2 := uint64(0)
+	parseUint := func(s string) uint64 {
+		id, _ := strconv.ParseUint(s, 10, 64)
+		return id
+	}
```

Comment:
You can use the `cast.ToUint64()` (the one without the error) to get rid of this custom bit